tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/utils/GitOperationWrapper.ts

```diff
@@ -0,0 +1,65 @@
+import { spawn } from 'child_process';
+import * as os from 'os';
+import simpleGit, { SimpleGit } from 'simple-git';
+import { Logger } from 'winston';
+
+export class GitOperationWrapper {
```

Comment:
Do we need to have these coded inside of the cli tool? Can we instead use the git cli directly in a shell script (sh or ps)?

Reply:
I don't necessarily have an issue with simple-git, I'm mostly just trying to have us think about whether or not we should be doing these operations directly in code or if they should just be standard git commands in a shell script in pipeline. If we need to add additional logic around these steps then maybe doing them inside this cli tool makes sense but if they are easily done in a pipeline step we should consider doing them in the shell script instead of writing this wrapper code.

Reply:
As this work is mainly for the docker experience, and we also want the service team to try docker at those own development environment, so my understanding is we cannot seperate this part into the pipeline. @dw511214992, I will let you comment on the ask from Wes.

Reply:
This wrapper is used by the task engine workflow of docker container, so we cannnot separate this part in

Reply:
As I mention I'm fine if we think this is useful here. However even if it isn't running in the pipeline it can still be done in a shell script inside of the task engine depending on how it is implemented.